### PR TITLE
HIVE-25529: Test reading/writing V2 tables with delete files

### PR DESCRIPTION
**Pull request by marton-bod**
- Add a couple of new helper methods to TestHelper and TestTables so we can easily create tables with V2 (formatVersion=2)
- Create new TestHiveIcebergV2 test suite for testing reads/writes to V2 tables with deletes
- Create new test util methods to create equality and position delete files in tests

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/TestHelper.java [hunk 1]
```diff
@@ -74,8 +82,10 @@ public Table table() {
   }
 
   public Map<String, String> properties() {
-    return ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, fileFormat.name(),
-        TableProperties.ENGINE_HIVE_ENABLED, "true");
+    Map<String, String> props = new HashMap<>(tblProps);
```
Comment (pvary) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/TestHelper.java:85]: Ryan prefers:
```
Maps.newHashMap(tblProps);
```

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java [hunk 2]
```diff
@@ -299,4 +311,68 @@ public static void validateDataWithSQL(TestHiveShell shell, String tableName, Li
...
+   * @param table The table to create the delete file for
+   * @param deleteFilePath The path where the delete file should be created, relative to the table location root
+   * @param equalityFields List of field names that should play a role in the equality check
+   * @param fileFormat The file format that should be used for writing out the delete file
+   * @param rowsToDelete The rows that should be deleted. It's enough to fill out the fields that are relevant for the
+   *                     equality check, as listed in equalityFields, the rest of the fields are ignored
+   * @return The DeleteFile created
+   * @throws IOException If there is an error during DeleteFile write
+   */
+  public static DeleteFile createEqualityDeleteFile(Table table, String deleteFilePath, List<String> equalityFields,
```
Comment (pvary) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:325]: The delete id = 3 could be written as either of the following equality delete files:

equality_ids=[1]

 1: id | -
-------|-
 3     | -        

equality_ids=[1]

Or:

 1: id | 2: category | 3: name
-------|-------------|---------
 3     | NULL        | Grizzly

We might want to test both

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java [hunk 3]
```diff
@@ -299,4 +311,68 @@ public static void validateDataWithSQL(TestHiveShell shell, String tableName, Li
...
+   * @param table The table to create the delete file for
+   * @param deleteFilePath The path where the delete file should be created, relative to the table location root
+   * @param fileFormat The file format that should be used for writing out the delete file
+   * @param partitionValues A map of partition values (partitionKey=partitionVal, ...) to be used for the delete file
+   * @param deletes The list of position deletes, each containing the data file path, the position of the row in the
+   *                data file and the row itself that should be deleted
+   * @return The DeleteFile created
+   * @throws IOException If there is an error during DeleteFile write
+   */
+  public static DeleteFile createPositionalDeleteFile(Table table, String deleteFilePath, FileFormat fileFormat,
```
Comment (pvary) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:356]: The positional deletes might contain the row data as well.
We might want to test that as well

Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:356]: It's currently tested with the rows present in the PositionalDelete. 
When I tried without supplying the rows, I got an NPE.
Interestingly, as long as you provide **_any_** row, it works, it doesn't need to be the actual row, e.g.
`new PositionDelete<Record>().set(dataFile.path(), 0L, GenericRecord.create(tbl.schema()))`
-> this works correctly as well to filter out the row at position 0

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java (hunk 2 above)
Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:325]: I think I'm covering that by leaving the last_name field as null, passing this in:
`add(1L, "Bob", null)`
i.e. not filling out the irrelevant fields.
Or is there something different you're looking for?

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/TestHelper.java (hunk 1 above)
Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/TestHelper.java:85]: Sure

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java (hunk 2 above)
Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:325]: Added a new test which omits a column that is not part of the equality check

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java (hunk 3 above)
Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:356]: Added a new test where the rows are not supplied. 
What caused the NPE was that I always supplied `table.schema()` as the positionDeleteSchema to the writer, even if the row was not available in the PositionDelete object.

Comment (pvary) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/HiveIcebergTestUtils.java:356]: Thx for the investigation

**Review by pvary — APPROVED** [review]

**Pull request merged**